File: LeetCode/日常刷题/201-400/299. 猜数字游戏.java
class Solution {
    public String getHint(String secret, String guess) {
        int n = secret.length();
        int bulls = 0;
        int[] secretCnt = new int[10];
        int[] guessCnt = new int[10];
        for (int i = 0; i < n; i++) {
            if (secret.charAt(i) == guess.charAt(i)) {
                bulls++;
            } else {
                secretCnt[secret.charAt(i) - '0']++;
                guessCnt[guess.charAt(i) - '0']++;
            }
        }
        int cows = 0;
        for (int i = 0; i < 10; i++) {
            cows += Math.min(secretCnt[i], guessCnt[i]);
        }
        return bulls + "A" + cows + "B";
    }
}

File: LeetCode/日常刷题/201-400/299. 猜数字游戏.py
from collections import Counter


class Solution:
    def getHint(self, secret: str, guess: str) -> str:
        secret_counter = Counter(secret)
        guess_counter = Counter(guess)
        bull_cnt = 0
        cow_cnt = 0
        for i in range(min(len(guess), len(secret))):
            if (secret[i] == guess[i]):
                bull_cnt += 1
                secret_counter[secret[i]] -= 1
                guess_counter[secret[i]] -= 1
        for k, cnt in guess_counter.items():
            cow_cnt += min(cnt, secret_counter[k])
        return f"{bull_cnt}A{cow_cnt}B"
